mono/io-layer/processes.c

```diff
@@ -1788,6 +1788,7 @@ gboolean EnumProcessModules (gpointer process, gpointer *modules,
 	if (!proc_name) {
 		modules[0] = NULL;
 		*needed = sizeof(gpointer);
+		g_free (proc_name);
```

Comment:
Here, proc_name is NULL (as per the if condition just above), so nothing to free